diff -rupN xmp_toolkit/common/XMP_LibUtils.hpp xmp_toolkit_tmp/common/XMP_LibUtils.hpp
--- xmp_toolkit/common/XMP_LibUtils.hpp 2010-06-17 17:21:52.000000000 +0200
+++ xmp_toolkit_tmp/common/XMP_LibUtils.hpp     2013-08-05 22:11:57.128515199 +0200
@@ -9,6 +9,7 @@
 // of the Adobe license agreement accompanying it.
 // =================================================================================================
 
+#include <stdio.h>
 #include "XMP_Environment.h"   // ! Must be the first include.
 #include "XMP_Const.h"
 
diff -rupN xmp_toolkit/XMPFiles/FileHandlers/AVCHD_Handler.cpp xmp_toolkit_tmp/XMPFiles/FileHandlers/AVCHD_Handler.cpp
--- xmp_toolkit/XMPFiles/FileHandlers/AVCHD_Handler.cpp 2010-06-17 17:21:52.000000000 +0200
+++ xmp_toolkit_tmp/XMPFiles/FileHandlers/AVCHD_Handler.cpp     2013-08-05 22:12:32.008515890 +0200
@@ -7,6 +7,7 @@
 // of the Adobe license agreement accompanying it.
 // =================================================================================================
 
+#include <cstdio>
 #include "AVCHD_Handler.hpp"
 
 #include "MD5.h"
